From vijaykumar@bravegnu.org  Mon Oct 26 15:03:10 2009
From: vijaykumar@bravegnu.org
Date: Mon, 21 Sep 2009 11:23:56 +0530
Subject: Staging: poch: Increase groups per interrupt
To: greg@kroah.com
Cc: jayakumar.lkml@gmail.com, alexey.zaytsev@gmail.com
Message-ID: <20090921055914.052071667@bravegnu.org>

From: Vijay Kumar B. <vijaykumar@bravegnu.org>

Increase groups per interrupt to reduce hogging of the CPU.

Signed-off-by: Vijay Kumar B. <vijaykumar@bravegnu.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/poch/poch.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/poch/poch.c
+++ b/drivers/staging/poch/poch.c
@@ -483,7 +483,7 @@ static void channel_dma_init(struct chan
 		  fpga + block_count_reg);
 	iowrite32(channel->group_count, fpga + group_count_reg);
 	/* FIXME: Hardcoded groups per int. Get it from sysfs? */
-	iowrite32(1, fpga + groups_per_int_reg);
+	iowrite32(16, fpga + groups_per_int_reg);
 
 	/* Unlock PCI address? Not defined in the data sheet, but used
 	 * in the reference code by Redrapids.
